### PR TITLE
WebAPI should notify when trace is received that matches filter

### PR DESCRIPTION
**Pull request by DoubleTK**

### CODE REVIEW

File: src/OddDotNet/Span.cs [hunk 1]
```diff
@@ -0,0 +1,73 @@
+namespace OddDotNet;
+
+public class Span
+{
+    public required Scope Scope { get; set; }
+    public required string Name { get; set; }
+    public required Dictionary<string, object> Attributes { get; set; }
+    public List<SpanEvent>? Events { get; set; }
```
Comment (DoubleTK) [inline src/OddDotNet/Span.cs:8]: All of these `Collection<T>?` should be `Collection<T>`. The collection should be initialized, and only a getter should be provided, no setter.

File: src/OddDotNet/Span.cs [hunk 2]
```diff
@@ -0,0 +1,73 @@
+namespace OddDotNet;
+
+public class Span
+{
+    public required Scope Scope { get; set; }
+    public required string Name { get; set; }
+    public required Dictionary<string, object> Attributes { get; set; }
```
Comment (DoubleTK) [inline src/OddDotNet/Span.cs:7]: Not `required`, no `init` or `set`, and init to empty.

File: src/OddDotNet/Span.cs (hunk 1 above)
Comment (DoubleTK) [inline src/OddDotNet/Span.cs:8]: Not nullable, no `init` or `set`, and init to empty.

File: src/OddDotNet/Span.cs [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    public List<SpanEvent>? Events { get; set; }
+    public uint Flags { get; set; }
```
Comment (DoubleTK) [inline src/OddDotNet/Span.cs:9]: `required`, change to `init`.

File: src/OddDotNet/Span.cs [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    public SpanKind Kind { get; set; } = SpanKind.Internal;
```
Comment (DoubleTK) [inline src/OddDotNet/Span.cs:10]: `required`, change to `init`, don't initialize.

File: src/OddDotNet/Span.cs [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+    public List<SpanLink>? Links { get; set; }
```
Comment (DoubleTK) [inline src/OddDotNet/Span.cs:11]: Not nullable, no `init` or `set`, and init to empty.

File: src/OddDotNet/Span.cs [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+    public SpanStatus Status { get; set; } = new SpanStatus { Code = SpanStatusCode.Unset };
```
Comment (DoubleTK) [inline src/OddDotNet/Span.cs:12]: `required`, change to `init`, don't initialize.

File: src/OddDotNet/Span.cs [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+    public required byte[] SpanId { get; set; }
+    public required byte[] TraceId { get; set; }
+    public required string TraceState { get; set; }
+    public byte[]? ParentSpanId { get; set; }
+    public ulong StartTimeUnixNano { get; set; }
```
Comment (DoubleTK) [inline src/OddDotNet/Span.cs:17]: `required` and `init`

File: src/OddDotNet/Span.cs [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+    public ulong EndTimeUnixNano { get; set; }
+}
+
+public class SpanStatus
+{
+    public SpanStatusCode Code { get; set; }
```
Comment (DoubleTK) [inline src/OddDotNet/Span.cs:23]: `required` and `init`

File: src/OddDotNet/Span.cs [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+    public string? Message { get; set; }
```
Comment (DoubleTK) [inline src/OddDotNet/Span.cs:24]: `init`.

File: src/OddDotNet/Span.cs [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+{
+    Unset = 0,
+    Ok = 1,
+    Error = 2
+}
+
+public class SpanLink
+{
+    public required byte[] SpanId { get; set; }
+    public Dictionary<string, object>? Attributes { get; set; }
```
Comment (DoubleTK) [inline src/OddDotNet/Span.cs:37]: Not nullable, no `init` or `set`, init to empty.

File: src/OddDotNet/Span.cs [hunk 11]
```diff
@@ -0,0 +1,73 @@
...
+    public uint Flags { get; set; }
```
Comment (DoubleTK) [inline src/OddDotNet/Span.cs:38]: `required` and `init`.

File: src/OddDotNet/Span.cs [hunk 12]
```diff
@@ -0,0 +1,73 @@
...
+    public byte[]? TraceId { get; set; }
```
Comment (DoubleTK) [inline src/OddDotNet/Span.cs:39]: `required`, not nullable, `init`, set to `[]`.

File: src/OddDotNet/Span.cs [hunk 13]
```diff
@@ -0,0 +1,73 @@
...
+    Internal = 1,
+    Server = 2,
+    Client = 3,
+    Producer = 4,
+    Consumer = 5
+}
+
+public class SpanEvent
+{
+    public required string Name { get; set; }
```
Comment (DoubleTK) [inline src/OddDotNet/Span.cs:55]: `init`

File: src/OddDotNet/Span.cs [hunk 14]
```diff
@@ -0,0 +1,73 @@
...
+    public required Dictionary<string, object> Attributes { get; set; }
```
Comment (DoubleTK) [inline src/OddDotNet/Span.cs:56]: Not `required`, no `init` or `set`, init to empty.

File: src/OddDotNet/Span.cs [hunk 15]
```diff
@@ -0,0 +1,73 @@
...
+    public uint TimeUnixNano { get; set; }
```
Comment (DoubleTK) [inline src/OddDotNet/Span.cs:57]: `required`, `init`

File: src/OddDotNet/Span.cs [hunk 16]
```diff
@@ -0,0 +1,73 @@
...
+    public string? Name { get; set; }
+    public Dictionary<string, object>? Attributes { get; set; }
+    public string? Version { get; set; }
+    public string? SchemaUrl { get; set; }
+    public required Resource Resource { get; set; }
+}
+
+public class Resource
+{
+    public required Dictionary<string, object> Attributes { get; set; }
```
Comment (DoubleTK) [inline src/OddDotNet/Span.cs:71]: Not `required`, no `init` or `set`, init to empty.

File: src/OddDotNet/Span.cs [hunk 17]
```diff
@@ -0,0 +1,73 @@
...
+{
+    public required string Name { get; set; }
+    public required Dictionary<string, object> Attributes { get; set; }
+    public uint TimeUnixNano { get; set; }
+}
+
+public class Scope
+{
+    public string? Name { get; set; }
+    public Dictionary<string, object>? Attributes { get; set; }
```
Comment (DoubleTK) [inline src/OddDotNet/Span.cs:63]: Not nullable, no `init` or `set`, init to empty.

File: src/OddDotNet/Span.cs [hunk 18]
```diff
@@ -0,0 +1,73 @@
...
+    public string? Version { get; set; }
+    public string? SchemaUrl { get; set; }
```
Comment (DoubleTK) [inline src/OddDotNet/Span.cs:65]: `init`.

File: src/OddDotNet/Span.cs [hunk 19]
```diff
@@ -0,0 +1,73 @@
...
+public class SpanEvent
+{
+    public required string Name { get; set; }
+    public required Dictionary<string, object> Attributes { get; set; }
+    public uint TimeUnixNano { get; set; }
+}
+
+public class Scope
+{
+    public string? Name { get; set; }
```
Comment (DoubleTK) [inline src/OddDotNet/Span.cs:62]: `required`, not null, `init`

File: src/OddDotNet/Span.cs [hunk 20]
```diff
@@ -0,0 +1,73 @@
...
+    public Dictionary<string, object>? Attributes { get; set; }
+    public string? Version { get; set; }
```
Comment (DoubleTK) [inline src/OddDotNet/Span.cs:64]: `init`

File: src/OddDotNet/Span.cs [hunk 21]
```diff
@@ -0,0 +1,73 @@
...
+    public string? SchemaUrl { get; set; }
+    public required Resource Resource { get; set; }
+}
+
+public class Resource
+{
+    public required Dictionary<string, object> Attributes { get; set; }
+    public string? SchemaUrl { get; set; }
```
Comment (DoubleTK) [inline src/OddDotNet/Span.cs:72]: `init`.

File: src/OddDotNet/Span.cs [hunk 22]
```diff
@@ -0,0 +1,73 @@
...
+public enum SpanStatusCode
+{
+    Unset = 0,
+    Ok = 1,
+    Error = 2
+}
+
+public class SpanLink
+{
+    public required byte[] SpanId { get; set; }
```
Comment (DoubleTK) [inline src/OddDotNet/Span.cs:36]: `init`

File: src/OddDotNet/Span.cs (hunk 1 above)
Comment (handro1) [reply src/OddDotNet/Span.cs]: Done

File: src/OddDotNet/Span.cs (hunk 2 above)
Comment (handro1) [reply src/OddDotNet/Span.cs]: Done

**Pull request merged**